### ISSUE TITLE
Large vertical images in galleries

### ISSUE THREAD

**Issue by alphachung** [dxprog/reddit-booru#39]
Currently in a gallery, if the height of an image is larger than the height of the viewport, it is not scaled, meaning that the user have to scroll to view the entire image, and the image can never be viewed in its entirety on the screen. 

![redditbooru-before](https://cloud.githubusercontent.com/assets/5276387/22406100/3dba8982-e687-11e6-99f2-d564dea4d00b.png)
<br><br>
I propose scaling those images in CSS, or other methods, such that the user can see the entire image at once, without scrolling. 

![redditbooru-after](https://cloud.githubusercontent.com/assets/5276387/22406104/41d85a1c-e687-11e6-9e23-e23af6911d2b.png)

**Comment by alphachung** [comment]
For comic strips, I propose using server-side scripts or client-side JavaScript to detect the aspect ratio of the image. Such that if the aspect ratio of the image is detected to be smaller than a certain value, then the CSS property or the selected method of scaling would not be applied in that case.

**Comment by dxprog** [comment]
Commit 8a10fff adds this functionality in a somewhat rudimentary fashion. I'd appreciate it if you could test it out and give me feedback: http://beta.redditbooru.com/gallery/9856/

**Comment by alphachung** [comment]
Other than I noticed that the first image is too close to the line of text containing the curator's name, I think the problem I noticed before is solved. 

However, IMHO the "expanded images" button might not be accessible to new users because it's not particularly noticeable at the bottom right corner. 

Maybe before the images there could be a line of text saying how many images there are in the gallery, and the button could be placed there instead? But I'm not sure that you want to deal with this in this issue though.

**Comment by dxprog** [comment]
Yeah, the presentation needs cleaning up for sure. I wanted to get feedback on the interaction before I went much further. It was shockingly more complex to get this working than I'd have thought, making me realize that the whole stack is a mess.

To the point of album numbers, that's something that I've been meaning to add back for a while (redditbooru v1 had that). I'll go ahead and open up a new ticket for that since it is a separate issue.

I'll get the presentation cleaned up tonight and ping you again for feedback if you're willing. Thanks!

**Comment by alphachung** [comment]
Sure
